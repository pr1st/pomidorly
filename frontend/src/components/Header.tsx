import React from 'react';
import "bootstrap/dist/css/bootstrap.css"
import classNames from "classnames";

type Props = {
    toMain: () => void,
    toStatistics: () => void,
    toSignUp: () => void,
    toSignIn: () => void,
    userName: string,
    logOut: () => void
}

const Header = (props: Props) => {
    const {toMain, toStatistics, toSignIn, toSignUp, userName, logOut} = props;
    const btnClass = classNames("btn", "btn-success", "btn-lg");
    let login;
    if (userName === "") {
        login = (
            <div className={classNames("Login")}>
                <button className={btnClass}
                        onClick={toSignIn}>
                    Sign in
                </button>
                <button className={btnClass}
                        onClick={toSignUp}>
                    Sign up
                </button>
            </div>
        )
    } else {
        login = (
            <div className={classNames("Login")}>
                <p>
                    {userName}
                </p>
                <button className={btnClass}
                        onClick={logOut}>
                    Log out
                </button>
            </div>
        )
    }
    return (
        <div className={classNames("Header")}>
            <div className={classNames("Page")}>
                <button className={btnClass}
                        onClick={toMain}
                        disabled={userName === ""}>
                    Main
                </button>
                <button className={btnClass}
                        onClick={toStatistics}
                        disabled={userName === ""}>
                    Statistics
                </button>
            </div>
            {login}
        </div>
    )
};

export default Header;