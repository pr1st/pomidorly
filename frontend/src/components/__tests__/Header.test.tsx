import React from 'react';
import renderer from 'react-test-renderer';
import Header from "../Header";

describe("Header component test", () => {
    it("renders correctly", () => {
        const dummy = () => {};
        const tree = renderer
            .create(
                <Header
                    logOut={dummy}
                    toMain={dummy}
                    toSignIn={dummy}
                    toSignUp={dummy}
                    toStatistics={dummy}
                    userName={"CustomUser"}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});