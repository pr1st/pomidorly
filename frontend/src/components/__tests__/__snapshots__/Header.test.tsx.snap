// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test renders correctly 1`] = `
<div
  className="Header"
>
  <div
    className="Page"
  >
    <button
      className="btn btn-success btn-lg"
      disabled={false}
      onClick={[Function]}
    >
      Main
    </button>
    <button
      className="btn btn-success btn-lg"
      disabled={false}
      onClick={[Function]}
    >
      Statistics
    </button>
  </div>
  <div
    className="Login"
  >
    <p>
      CustomUser
    </p>
    <button
      className="btn btn-success btn-lg"
      onClick={[Function]}
    >
      Log out
    </button>
  </div>
</div>
`;
